app/services/mentee_final_survey_form.rb

```diff
@@ -24,6 +24,9 @@ def initialize(attrs, user)
   def save
     survey = FinalSurvey.find_or_initialize_by(project_id: @user.project.id)
     survey.update_attributes(attributes.merge({mentee_id: @user.id, edition_id: @user.id}))
+    user = User.find(@user.id)
+    # probably would be best to replace it with num constants after inserting all badges
+    user.badges << Badge.where(["name = ?", "Final survey"]).first 
```

Comment:
you can use `find_by(name: "Final survey")`. But I agree it's better to use number here. Let's use enum for badges. Name might change at some point so it will be less troublesome to keep track of it